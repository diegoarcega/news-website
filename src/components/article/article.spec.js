import React from 'react'
import Article from './article'
import { render, fireEvent } from '@testing-library/react'

it('renders the Article with ImagePlaceHolder', () => {
  const { asFragment } = render(
    <Article
      title="Article"
      img="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg=="
      content="This article is informative"
      author="Diego Arcega"
    />
  )
  expect(asFragment()).toMatchSnapshot('with-loading')
})

it('renders the Article without ImagePlaceHolder', async () => {
  const { getByTestId, queryByTestId } = render(
    <Article
      title="Article"
      img="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg=="
      content="This article is informative"
      author="Diego Arcega"
    />
  )

  fireEvent.load(getByTestId('article-image'))
  const imagePlaceholder = queryByTestId('image-placeholder')
  expect(imagePlaceholder).toBeNull()
})
