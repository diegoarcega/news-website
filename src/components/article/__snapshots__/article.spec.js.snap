// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Article with ImagePlaceHolder: with-loading 1`] = `
<DocumentFragment>
  .c0 {
  padding: 10px 0 10px 0;
  border-radius: 4px;
}

.c0:hover {
  cursor: pointer;
}

.c2 {
  padding-top: 13px;
}

.c1 {
  display: none;
  max-width: 100%;
  max-height: 200px;
  height: 100%;
  border-radius: 4px;
}

.c3 {
  padding-top: 10px;
}

.c4 {
  padding-top: 10px;
}

@media only screen and (min-width:1rem) {
  .c1 {
    width: 100%;
  }
}

@media only screen and (min-width:48rem) {
  .c1 {
    width: initial;
  }
}

<article
    class="c0"
  >
    <img
      class="c1"
      data-testid="article-image"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg=="
    />
    <svg
      aria-labelledby="Loading interface..."
      data-testid="image-placeholder"
      preserveAspectRatio="none"
      role="img"
      viewBox="0 0 330 197"
    >
      <title>
        Loading interface...
      </title>
      <rect
        clip-path="url(#i)"
        height="197"
        style="fill: url(#i);"
        width="330"
        x="0"
        y="0"
      />
      <defs>
        <clippath
          id="i"
        >
          <rect
            height="200"
            rx="5"
            ry="5"
            width="330"
            x="0"
            y="0"
          />
        </clippath>
        <lineargradient
          id="i"
        >
          <stop
            offset="0%"
            stop-color="#f0f0f0"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="1s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="-2; -2; 1"
            />
          </stop>
          <stop
            offset="50%"
            stop-color="#e0e0e0"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="1s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="-1; -1; 2"
            />
          </stop>
          <stop
            offset="100%"
            stop-color="#f0f0f0"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="1s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="0; 0; 3"
            />
          </stop>
        </lineargradient>
      </defs>
    </svg>
    <h1
      class="c2"
    >
      Article
    </h1>
    <small
      class="c3"
    >
      Diego Arcega
    </small>
    <p
      class="c4"
    >
      This article is informative
    </p>
  </article>
</DocumentFragment>
`;
